File: app/src/main/java/com/sgif/makegif/view/photo/PhotoAdapter.java
package com.sgif.makegif.view.photo;

import android.content.Context;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.sgif.makegif.R;
import com.sgif.makegif.common.GlideApp;
import com.sgif.makegif.common.base.BaseAdapter;
import com.sgif.makegif.common.base.BaseViewHolder;
import com.sgif.makegif.domain.model.Photo;
import com.sgif.makegif.util.Utils;

import java.io.File;


public class PhotoAdapter extends BaseAdapter<Photo, PhotoAdapter.ItemViewHolder> {
    public interface OnClickItemPhotoListener {
        void onClickItemPhoto(Photo photo);
    }

    private final int mSizeImage;
    private OnClickItemPhotoListener mOnClickItemHomeListener;

    public void setOnClickItemPhotoListener(OnClickItemPhotoListener onClickItemHomeListener) {
        this.mOnClickItemHomeListener = onClickItemHomeListener;
    }

    PhotoAdapter(Context context) {
        super(context);
        mSizeImage = (Utils.getScreenWidth(mContext) - 2 * Utils.convertDpToPixel(mContext, 5)) / 3;
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_photo, parent, false);
        return new ItemViewHolder(view);
    }

    @Override
    public int getItemCount() {
        return size();
    }

    public class ItemViewHolder extends BaseViewHolder<Photo> implements View.OnClickListener {

        private ImageView mImgPhoto;

        ItemViewHolder(View itemView) {
            super(itemView);
            mImgPhoto = itemView.findViewById(R.id.imgPhoto);

            mImgPhoto.getLayoutParams().height = mSizeImage;
            mImgPhoto.getLayoutParams().width = mSizeImage;

            itemView.setOnClickListener(this);
        }

        @Override
        public void bindData(Photo photo) {
            String link = photo.getPath();
            if (link != null) {
                GlideApp
                        .with(mContext)
                        .load(new File(link))
                        .centerCrop()
                        .override(mSizeImage, mSizeImage)
                        .into(mImgPhoto);
            }
        }

        @Override
        public void onClick(View v) {
            if (mOnClickItemHomeListener != null) {
                mOnClickItemHomeListener.onClickItemPhoto(getItem(getLayoutPosition()));
            }
        }
    }
}

File: app/src/main/java/com/quangtd/qgifmaker/screen/new_gallery/adapter/PhotoAdapter.kt
package com.quangtd.qgifmaker.screen.new_gallery.adapter;

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.quangtd.qgifmaker.R
import com.quangtd.qgifmaker.common.base.BaseAdapter
import com.quangtd.qgifmaker.common.base.BaseViewHolder
import com.quangtd.qgifmaker.domain.model.Photo
import com.quangtd.qgifmaker.util.ScreenUtils
import kotlinx.android.synthetic.main.item_photo.view.*


/**
 * Created by QuangTD
 * on 1/18/2018.
 */
class PhotoAdapter(private val mContext: Context) : BaseAdapter<Photo, PhotoAdapter.PhotoViewHolder>() {

    private var mPhotoDimen: Int = (ScreenUtils.getWidthScreen(mContext) - 2 * ScreenUtils.convertDpToPixel(mContext, 5)) / 2
    var onClickPhotoListener: OnClickPhotoListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PhotoViewHolder {
        return PhotoViewHolder(LayoutInflater.from(mContext).inflate(R.layout.item_photo, parent, false))
    }

    inner class PhotoViewHolder(itemView: View) : BaseViewHolder<Photo>(itemView) {
        private val imvThumbnail: ImageView = itemView.imvThumbnail

        init {
            imvThumbnail.layoutParams.height = mPhotoDimen
            imvThumbnail.layoutParams.width = mPhotoDimen
        }

        override fun bindData(t: Photo) {
            Glide.with(mContext)
                    .load(if (t.uri != null) t.uri else t.path)
                    .apply(RequestOptions()
                            .placeholder(R.drawable.placeholder)
                            .centerCrop()
                            .error(R.drawable.placeholder))
                    .into(imvThumbnail)
            imvThumbnail.setOnClickListener {
                onClickPhotoListener?.onClickPhoto(t)
            }
        }
    }

    interface OnClickPhotoListener {
        fun onClickPhoto(photo: Photo)
    }
}